### PR TITLE
Commited after implementing following-follwers

### PR DESCRIPTION
**Pull request by manazirahsan**

### CODE REVIEW

File: app/assets/javascripts/users.js.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+$(function() {
+    var element = $('#follow_unfollow');
+    element.click(function() {
+        alert('Hi')
```
Comment (chuanhtuan) [inline app/assets/javascripts/users.js.erb]: remove

File: app/assets/stylesheets/relationships.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Relationships controller here.
```
Comment (chuanhtuan) [inline app/assets/stylesheets/relationships.scss]: remove unused file

File: app/helpers/relationships_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module RelationshipsHelper
```
Comment (chuanhtuan) [inline app/helpers/relationships_helper.rb]: remove unused file

File: app/models/user.rb [hunk 4]
```diff
@@ -37,4 +45,21 @@ def remember
...
+  end
+
+  # Unfollows a user.
+  def unfollow(other_user)
+    active_relationships.find_by(followed_id: other_user.id).destroy
+  end
+
+  # Returns true if the current user is following the other user.
+  def following?(other_user)
+    following.include?(other_user)
```
Comment (chuanhtuan) [inline app/models/user.rb]: ()

File: app/models/user.rb [hunk 5]
```diff
@@ -37,4 +45,21 @@ def remember
...
+    active_relationships.create(followed_id: other_user.id)
+  end
+
+  # Unfollows a user.
+  def unfollow(other_user)
+    active_relationships.find_by(followed_id: other_user.id).destroy
+  end
+
+  # Returns true if the current user is following the other user.
+  def following?(other_user)
```
Comment (chuanhtuan) [inline app/models/user.rb]: ()

File: app/models/user.rb [hunk 6]
```diff
@@ -37,4 +45,21 @@ def remember
   def forget
     update_attributes! remember_digest: nil
   end
+
+  # Follows a user.
+  def follow(other_user)
```
Comment (chuanhtuan) [inline app/models/user.rb]: ()

File: app/models/user.rb [hunk 7]
```diff
@@ -37,4 +45,21 @@ def remember
...
+    active_relationships.create(followed_id: other_user.id)
+  end
+
+  # Unfollows a user.
+  def unfollow(other_user)
```
Comment (chuanhtuan) [inline app/models/user.rb]: ()

File: app/models/user.rb [hunk 8]
```diff
@@ -37,4 +45,21 @@ def remember
   def forget
     update_attributes! remember_digest: nil
   end
+
+  # Follows a user.
+  def follow(other_user)
+    active_relationships.create(followed_id: other_user.id)
```
Comment (chuanhtuan) [inline app/models/user.rb]: ()

File: app/views/users/show.html.erb [hunk 9]
```diff
@@ -6,6 +6,13 @@
         <%= gravatar_for @user %>
         <%= @user.name %>
       </h1>
+      <% if @user != current_user %>
+          <% if current_user.following.include?(@user) %>
+              <%= link_to "Unfollow", {controller: 'relationships', action: 'destroy', id: @user.id}, method: :delete, class: 'btn btn-xs btn-danger' %>
```
Comment (chuanhtuan) [inline app/views/users/show.html.erb]: <%= link_to "Unfollow", relationship_path(id: @user.id), method: :delete, class: 'btn btn-xs btn-danger' %>

File: app/assets/javascripts/users.js.erb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+$(function() {
+    var element = $('#follow_unfollow');
+    element.click(function() {
+        $.get("/followUnfollow", { 'followUnfollowUserInstruction': element.html() }, function(data) {
```
Comment (chuanhtuan) [inline app/assets/javascripts/users.js.erb]: $.get("/followUnfollow",
followUnfollow path is not in routes config

File: app/views/users/show.html.erb [hunk 11]
```diff
@@ -6,6 +6,13 @@
         <%= gravatar_for @user %>
         <%= @user.name %>
       </h1>
+      <% if @user != current_user %>
+          <% if current_user.following.include?(@user) %>
+              <%= link_to "Unfollow", relationship_path(id: @user.id), method: :delete, class: 'btn btn-xs btn-danger' %>
```
Comment (chuanhtuan) [inline app/views/users/show.html.erb]: i18n

File: app/views/users/show.html.erb [hunk 12]
```diff
@@ -6,6 +6,13 @@
...
+          <% else %>
+              <%= link_to "Follow", {controller: 'relationships', action: 'create', id: @user.id}, method: :post, class: 'btn btn-xs btn-info' %>
```
Comment (chuanhtuan) [inline app/views/users/show.html.erb]: same above

File: app/assets/javascripts/users.js.erb (hunk 10 above)
Comment (manazirahsan) [inline app/assets/javascripts/users.js.erb]: For the time being, we don't need this file, that's why I removed it.

**Pull request merged**